core/src/actors/chain_manager/mod.rs

```diff
@@ -146,6 +148,36 @@ impl ChainManager {
             .collect()
     }
 
+    /// Method to update utxo set
+    fn update_utxo_set(&self, block: Block) -> HashMap<OutputPointer, Output> {
+        // Create a copy of the state "unspent_outputs_pool"
+        let mut utxo_set = self.unspent_outputs_pool.clone();
+
+        let transactions = block.txns;
+
+        for transaction in transactions {
+            let txn_hash = transaction.hash();
+            for input in transaction.inputs {
+                // Obtain the OuputPointer of each input and remove it from the utxo_set
+                let output_pointer = get_output_pointer_from_input(&input);
+
+                utxo_set.remove(&output_pointer);
+            }
+
+            for (index, output) in transaction.outputs.iter().enumerate() {
+                // Add the new outputs to the utxo_set
+                let output_pointer = OutputPointer {
+                    transaction_id: txn_hash,
+                    output_index: index as u32,
+                };
+
+                utxo_set.insert(output_pointer, output.clone());
+            }
+        }
+
+        utxo_set
+    }
```

Comment:
The idea of this function is obtain a copy of the utxo_set with this blocks modifications. Because in a future we will handle different utxo_set and will only persist one of them.